From: Tommy Webb <tommy@calyxinstitute.org>
Date: Wed, 26 Jul 2023 16:57:20 -0400
Subject: Bring in Brave feature states for privacy/security

Now using cromite flags as much as possible.

Reviewed Brave's feature state overrides, as of brave-core tag v1.67.9,
and added any that were relevant for privacy or security and applicable
to Chromium 124. Features that were already set a particular way still
had comments added so that it will be easier to catch if upstream makes
unwanted changes to their states.

Testing feature changes (two examples):
1. AnonymousIframe: https://anonymous-iframe.glitch.me
   Will show the state enabled prior to this change, disabled after.
2. AutofillServerCommunication: Install and activate Rethink as VPN,
   clear browser storage, do a test search from the omnibar, and look
   for content-autofill.googleapis.com in Rethink logs. It will be
   present before this change, absent after.

brave-core can be found here: https://github.com/brave/brave-core

app/feature_defaults_unittest.cc is a highly useful resource,
and the work of their privacy/security team is greatly appreciated.

Change-Id: Ic0e67157ff4a747bb05ceba264660bfeb1ac73ef
---
 .../extension_telemetry_service_factory.cc    |  17 +-
 .../attribution_storage_delegate_impl.cc      |   3 +
 cromite_flags/BUILD.gn                        | 162 +++++++++++++++++-
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../common/aw_features_cc/placeholder.txt     |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../base/features_cc/placeholder.txt          |   1 +
 ...ve-feature-states-for-privacy-security.inc |   4 +
 ...ve-feature-states-for-privacy-security.inc |   3 +
 .../core/features_cc/placeholder.txt          |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 ...ve-feature-states-for-privacy-security.inc |   3 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../visual_query/features_cc/placeholder.txt  |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../features_cc/placeholder.txt               |   1 +
 ...ve-feature-states-for-privacy-security.inc |   2 +
 .../autofill_features_cc/placeholder.txt      |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../placeholder.txt                           |   1 +
 ...ve-feature-states-for-privacy-security.inc |   7 +
 .../commerce_feature_list_cc/placeholder.txt  |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../placeholder.txt                           |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../core/browser/features_cc/placeholder.txt  |   1 +
 ...ve-feature-states-for-privacy-security.inc |  11 ++
 .../core/features_cc/placeholder.txt          |   1 +
 ...ve-feature-states-for-privacy-security.inc |   2 +
 .../placeholder.txt                           |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../metrics_features_cc/placeholder.txt       |   1 +
 ...ve-feature-states-for-privacy-security.inc |   3 +
 .../placeholder.txt                           |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../network_time_tracker_cc/placeholder.txt   |   1 +
 ...ve-feature-states-for-privacy-security.inc |   5 +
 .../omnibox_features_cc/placeholder.txt       |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../placeholder.txt                           |   1 +
 ...ve-feature-states-for-privacy-security.inc |   2 +
 .../features_cc/placeholder.txt               |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../placeholder.txt                           |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 ...ve-feature-states-for-privacy-security.inc |   4 +
 .../placeholder.txt                           |   1 +
 ...ve-feature-states-for-privacy-security.inc |   2 +
 .../core/common/features_cc/placeholder.txt   |   1 +
 ...ve-feature-states-for-privacy-security.inc |   6 +
 .../search/ntp_features_cc/placeholder.txt    |   1 +
 ...ve-feature-states-for-privacy-security.inc |   3 +
 .../public/features_cc/placeholder.txt        |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../placeholder.txt                           |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../common/common_features_cc/placeholder.txt |   1 +
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../prefetch_features_cc/placeholder.txt      |   1 +
 ...ve-feature-states-for-privacy-security.inc |   2 +
 ...ve-feature-states-for-privacy-security.inc |   8 +
 ...ve-feature-states-for-privacy-security.inc |   3 +
 ...ve-feature-states-for-privacy-security.inc |   2 +
 ...ve-feature-states-for-privacy-security.inc |  18 ++
 ...ve-feature-states-for-privacy-security.inc |   1 +
 .../base/ui_base_features_cc/placeholder.txt  |   1 +
 .../renderer/core/html/html_script_element.cc |   4 +-
 .../platform/runtime_enabled_features.json5   |  42 ++---
 69 files changed, 332 insertions(+), 31 deletions(-)
 create mode 100644 cromite_flags/android_webview/common/aw_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/android_webview/common/aw_features_cc/placeholder.txt
 create mode 100644 cromite_flags/base/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/base/features_cc/placeholder.txt
 create mode 100644 cromite_flags/chrome/browser/browser_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100644 cromite_flags/chrome/browser/companion/core/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/chrome/browser/companion/core/features_cc/placeholder.txt
 create mode 100644 cromite_flags/chrome/browser/ui/ui_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100644 cromite_flags/chrome/common/chrome_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100644 cromite_flags/chrome/common/companion/visual_query/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/chrome/common/companion/visual_query/features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/aggregation_service/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/aggregation_service/features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/autofill/core/common/autofill_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/autofill/core/common/autofill_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/autofill/core/common/autofill_payments_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/autofill/core/common/autofill_payments_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/commerce/core/commerce_feature_list_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/commerce/core/commerce_feature_list_cc/placeholder.txt
 create mode 100644 cromite_flags/components/content_settings/core/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100644 cromite_flags/components/heap_profiling/in_process/heap_profile_parameters_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/heap_profiling/in_process/heap_profile_parameters_cc/placeholder.txt
 create mode 100644 cromite_flags/components/history/core/browser/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/history/core/browser/features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/history_clusters/core/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/history_clusters/core/features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/history_clusters/core/on_device_clustering_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/history_clusters/core/on_device_clustering_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/metrics/metrics_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/metrics/metrics_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/metrics/structured/structured_metrics_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/metrics/structured/structured_metrics_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/network_time/network_time_tracker_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/network_time/network_time_tracker_cc/placeholder.txt
 create mode 100644 cromite_flags/components/omnibox/common/omnibox_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/omnibox/common/omnibox_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/optimization_guide/core/optimization_guide_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/optimization_guide/core/optimization_guide_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/page_image_service/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/page_image_service/features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/password_manager/core/common/password_manager_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/password_manager/core/common/password_manager_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/permissions/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100644 cromite_flags/components/privacy_sandbox/privacy_sandbox_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/privacy_sandbox/privacy_sandbox_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/safe_browsing/core/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/safe_browsing/core/common/features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/search/ntp_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/search/ntp_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/segmentation_platform/public/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/segmentation_platform/public/features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/shared_highlighting/core/common/shared_highlighting_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/shared_highlighting/core/common/shared_highlighting_features_cc/placeholder.txt
 create mode 100644 cromite_flags/components/subresource_filter/core/common/common_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/components/subresource_filter/core/common/common_features_cc/placeholder.txt
 create mode 100644 cromite_flags/content/browser/preloading/prefetch/prefetch_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/content/browser/preloading/prefetch/prefetch_features_cc/placeholder.txt
 create mode 100644 cromite_flags/content/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100644 cromite_flags/content/public/common/content_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100644 cromite_flags/net/base/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100644 cromite_flags/services/network/public/cpp/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100644 cromite_flags/third_party/blink/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100644 cromite_flags/ui/base/ui_base_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
 create mode 100755 cromite_flags/ui/base/ui_base_features_cc/placeholder.txt

diff --git a/chrome/browser/safe_browsing/extension_telemetry/extension_telemetry_service_factory.cc b/chrome/browser/safe_browsing/extension_telemetry/extension_telemetry_service_factory.cc
--- a/chrome/browser/safe_browsing/extension_telemetry/extension_telemetry_service_factory.cc
+++ b/chrome/browser/safe_browsing/extension_telemetry/extension_telemetry_service_factory.cc
@@ -20,9 +20,13 @@
 
 namespace safe_browsing {
 
+// We attempt to turn off all of this, now that the kExtensionTelemetry flag is removed.
+// See brave-core 1d8262e0f94031472bb1419ddac1bdfbce6cbffb
+
 // static
 ExtensionTelemetryService* ExtensionTelemetryServiceFactory::GetForProfile(
     Profile* profile) {
+  if ((true)) return nullptr;
   return static_cast<ExtensionTelemetryService*>(
       GetInstance()->GetServiceForBrowserContext(profile, /* create= */ true));
 }
@@ -36,12 +40,13 @@ ExtensionTelemetryServiceFactory::GetInstance() {
 
 ExtensionTelemetryServiceFactory::ExtensionTelemetryServiceFactory()
     : ProfileKeyedServiceFactory("ExtensionTelemetryService",
-                                 ProfileSelections::BuildForRegularProfile()) {
-  DependsOn(NetworkContextServiceFactory::GetInstance());
-  DependsOn(extensions::ExtensionPrefsFactory::GetInstance());
-  DependsOn(extensions::ExtensionRegistryFactory::GetInstance());
-  DependsOn(extensions::ExtensionManagementFactory::GetInstance());
-  DependsOn(extensions::ExtensionSystemFactory::GetInstance());
+                                 ProfileSelections::BuildNoProfilesSelected()) {
+}
+
+content::BrowserContext*
+ExtensionTelemetryServiceFactory::GetBrowserContextToUse(
+    content::BrowserContext* context) const {
+  return nullptr;
 }
 
 std::unique_ptr<KeyedService>
diff --git a/content/browser/attribution_reporting/attribution_storage_delegate_impl.cc b/content/browser/attribution_reporting/attribution_storage_delegate_impl.cc
--- a/content/browser/attribution_reporting/attribution_storage_delegate_impl.cc
+++ b/content/browser/attribution_reporting/attribution_storage_delegate_impl.cc
@@ -237,6 +237,9 @@ AttributionStorageDelegateImpl::GetNullAggregatableReports(
     std::optional<base::Time> attributed_source_time) const {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
 
+  // See brave-core da32f5c44b1563e408eabe2f8e9ed263d28ce468
+  if ((true)) return {};
+
   switch (noise_mode_) {
     case AttributionNoiseMode::kDefault:
       return GetNullAggregatableReportsImpl(trigger, trigger_time,
diff --git a/cromite_flags/BUILD.gn b/cromite_flags/BUILD.gn
--- a/cromite_flags/BUILD.gn
+++ b/cromite_flags/BUILD.gn
@@ -18,6 +18,16 @@
 # this file is intentionally empty
 #
 
+cpp_bromite_include("android_webview_common_aw_features_cc") {
+  inputs = [ "//cromite_flags/android_webview/common/aw_features_cc/placeholder.txt" ]
+  output_file = "android_webview_common_aw_features_cc.inc"
+}
+
+cpp_bromite_include("base_features_cc") {
+  inputs = [ "//cromite_flags/base/features_cc/placeholder.txt" ]
+  output_file = "base_features_cc.inc"
+}
+
 cpp_bromite_include("chrome_browser_about_flags_cc") {
   inputs = [ "//cromite_flags/chrome/browser/about_flags_cc/placeholder.txt" ]
   output_file = "chrome_browser_about_flags_cc.inc"
@@ -28,6 +38,11 @@ cpp_bromite_include("chrome_common_chrome_features_cc") {
   output_file = "chrome_common_chrome_features_cc.inc"
 }
 
+cpp_bromite_include("chrome_common_companion_visual_query_features_cc") {
+  inputs = [ "//cromite_flags/chrome/common/companion/visual_query/features_cc/placeholder.txt" ]
+  output_file = "chrome_common_companion_visual_query_features_cc"
+}
+
 cpp_bromite_include("content_common_features_cc") {
   inputs = [ "//cromite_flags/content/common/features_cc/placeholder.txt" ]
   output_file = "content_common_features_cc.inc"
@@ -68,6 +83,11 @@ cpp_bromite_include("chrome_browser_browser_features_cc") {
   output_file = "chrome_browser_browser_features_cc.inc"
 }
 
+cpp_bromite_include("chrome_browser_companion_core_features_cc") {
+  inputs = [ "//cromite_flags/chrome/browser/companion/core/features_cc/placeholder.txt" ]
+  output_file = "chrome_browser_companion_core_features_cc.inc"
+}
+
 cpp_bromite_include("chrome_browser_ui_ui_features_cc") {
   inputs = [ "//cromite_flags/chrome/browser/ui/ui_features_cc/placeholder.txt" ]
   output_file = "chrome_browser_ui_ui_features_cc.inc"
@@ -83,14 +103,59 @@ cpp_bromite_include("media_base_media_switches_h") {
   output_file = "media_base_media_switches_h.inc"
 }
 
+cpp_bromite_include("components_aggregation_service_features_cc") {
+  inputs = [ "//cromite_flags/components/aggregation_service/features_cc/placeholder.txt" ]
+  output_file = "components_aggregation_service_features_cc.inc"
+}
+
+cpp_bromite_include("components_autofill_core_common_autofill_features_cc") {
+  inputs = [ "//cromite_flags/components/autofill/core/common/autofill_features_cc/placeholder.txt" ]
+  output_file = "components_autofill_core_common_autofill_features_cc.inc"
+}
+
+cpp_bromite_include("components_autofill_core_common_autofill_payments_features_cc") {
+  inputs = [ "//cromite_flags/components/autofill/core/common/autofill_payments_features_cc/placeholder.txt" ]
+  output_file = "components_autofill_core_common_autofill_payments_features_cc.inc"
+}
+
+cpp_bromite_include("components_commerce_core_commerce_feature_list_cc") {
+  inputs = [ "//cromite_flags/components/commerce/core/commerce_feature_list_cc/placeholder.txt" ]
+  output_file = "components_commerce_core_commerce_feature_list_cc.inc"
+}
+
 cpp_bromite_include("components_content_settings_core_common_features_cc") {
   inputs = [ "//cromite_flags/components/content_settings/core/common/features_cc/placeholder.txt" ]
   output_file = "components_content_settings_core_common_features_cc.inc"
 }
 
-cpp_bromite_include("components_permissions_features_cc") {
-  inputs = [ "//cromite_flags/components/permissions/features_cc/placeholder.txt" ]
-  output_file = "components_permissions_features_cc.inc"
+cpp_bromite_include("components_history_core_browser_features_cc") {
+  inputs = [ "//cromite_flags/components/history/core/browser/features_cc/placeholder.txt" ]
+  output_file = "components_history_core_browser_features_cc.inc"
+}
+
+cpp_bromite_include("components_history_clusters_core_features_cc") {
+  inputs = [ "//cromite_flags/components/history_clusters/core/features_cc/placeholder.txt" ]
+  output_file = "components_history_clusters_core_features_cc.inc"
+}
+
+cpp_bromite_include("components_history_clusters_core_on_device_clustering_features_cc") {
+  inputs = [ "//cromite_flags/components/history_clusters/core/on_device_clustering_features_cc/placeholder.txt" ]
+  output_file = "components_history_clusters_core_on_device_clustering_features_cc.inc"
+}
+
+cpp_bromite_include("components_metrics_metrics_features_cc") {
+  inputs = [ "//cromite_flags/components/metrics/metrics_features_cc/placeholder.txt" ]
+  output_file = "components_metrics_metrics_features_cc.inc"
+}
+
+cpp_bromite_include("components_metrics_structured_structured_metrics_features_cc") {
+  inputs = [ "//cromite_flags/components/metrics/structured/structured_metrics_features_cc/placeholder.txt" ]
+  output_file = "components_metrics_structured_structured_metrics_features_cc.inc"
+}
+
+cpp_bromite_include("components_network_time_network_time_tracker_cc") {
+  inputs = [ "//cromite_flags/components/network_time/network_time_tracker_cc/placeholder.txt" ]
+  output_file = "components_network_time_network_time_tracker_cc.inc"
 }
 
 cpp_bromite_include("components_offline_pages_core_offline_page_feature_cc") {
@@ -103,6 +168,41 @@ cpp_bromite_include("components_offline_pages_core_offline_page_feature_h") {
   output_file = "components_offline_pages_core_offline_page_feature_h.inc"
 }
 
+cpp_bromite_include("components_page_image_service_features_cc") {
+  inputs = [ "//cromite_flags/components/page_image_service/features_cc/placeholder.txt" ]
+  output_file = "components_page_image_service_features_cc.inc"
+}
+
+cpp_bromite_include("components_permissions_features_cc") {
+  inputs = [ "//cromite_flags/components/permissions/features_cc/placeholder.txt" ]
+  output_file = "components_permissions_features_cc.inc"
+}
+
+cpp_bromite_include("components_safe_browsing_core_common_features_cc") {
+  inputs = [ "//cromite_flags/components/safe_browsing/core/common/features_cc/placeholder.txt" ]
+  output_file = "components_safe_browsing_core_common_features_cc.inc"
+}
+
+cpp_bromite_include("components_search_ntp_features_cc") {
+  inputs = [ "//cromite_flags/components/search/ntp_features_cc/placeholder.txt" ]
+  output_file = "components_search_ntp_features_cc.inc"
+}
+
+cpp_bromite_include("components_segmentation_platform_public_features_cc") {
+  inputs = [ "//cromite_flags/components/segmentation_platform/public/features_cc/placeholder.txt" ]
+  output_file = "components_segmentation_platform_public_features_cc.inc"
+}
+
+cpp_bromite_include("components_shared_highlighting_core_common_shared_highlighting_features_cc") {
+  inputs = [ "//cromite_flags/components/shared_highlighting/core/common/shared_highlighting_features_cc/placeholder.txt" ]
+  output_file = "components_shared_highlighting_core_common_shared_highlighting_features_cc.inc"
+}
+
+cpp_bromite_include("components_subresource_filter_core_common_common_features_cc") {
+  inputs = [ "//cromite_flags/components/subresource_filter/core/common/common_features_cc/placeholder.txt" ]
+  output_file = "components_subresource_filter_core_common_common_features_cc.inc"
+}
+
 cpp_bromite_include("net_base_features_cc") {
   inputs = [ "//cromite_flags/net/base/features_cc/placeholder.txt" ]
   output_file = "net_base_features_cc.inc"
@@ -123,25 +223,81 @@ cpp_bromite_include("services_network_public_cpp_features_h") {
   output_file = "services_network_public_cpp_features_h.inc"
 }
 
+cpp_bromite_include("components_heap_profiling_in_process_heap_profile_parameters_cc") {
+  inputs = [ "//cromite_flags/components/heap_profiling/in_process/heap_profile_parameters_cc/placeholder.txt" ]
+  output_file = "components_heap_profiling_in_process_heap_profile_parameters_cc.inc"
+}
+
+cpp_bromite_include("components_omnibox_common_omnibox_features_cc") {
+  inputs = [ "//cromite_flags/components/omnibox/common/omnibox_features_cc/placeholder.txt" ]
+  output_file = "components_omnibox_common_omnibox_features_cc.inc"
+}
+
+cpp_bromite_include("components_optimization_guide_core_optimization_guide_features_cc") {
+  inputs = [ "//cromite_flags/components/optimization_guide/core/optimization_guide_features_cc/placeholder.txt" ]
+  output_file = "components_optimization_guide_core_optimization_guide_features_cc.inc"
+}
+
+cpp_bromite_include("components_password_manager_core_common_password_manager_features_cc") {
+  inputs = [ "//cromite_flags/components/password_manager/core/common/password_manager_features_cc/placeholder.txt" ]
+  output_file = "components_password_manager_core_common_password_manager_features_cc.inc"
+}
+
+cpp_bromite_include("components_privacy_sandbox_privacy_sandbox_features_cc") {
+  inputs = [ "//cromite_flags/components/privacy_sandbox/privacy_sandbox_features_cc/placeholder.txt" ]
+  output_file = "components_privacy_sandbox_privacy_sandbox_features_cc.inc"
+}
+
+cpp_bromite_include("content_browser_preloading_prefetch_prefetch_features_cc") {
+  inputs = [ "//cromite_flags/content/browser/preloading/prefetch/prefetch_features_cc/placeholder.txt" ]
+  output_file = "content_browser_preloading_prefetch_prefetch_features_cc.inc"
+}
+
 component("cromite_flags") {
   deps = [
+    ":android_webview_common_aw_features_cc",
+    ":base_features_cc",
+    ":content_browser_preloading_prefetch_prefetch_features_cc",
     ":content_common_features_cc",
     ":content_public_common_content_features_cc",
     ":content_public_common_content_features_h",
+    ":components_aggregation_service_features_cc",
+    ":components_autofill_core_common_autofill_features_cc",
+    ":components_autofill_core_common_autofill_payments_features_cc",
+    ":components_commerce_core_commerce_feature_list_cc",
     ":components_content_settings_core_common_features_cc",
+    ":components_heap_profiling_in_process_heap_profile_parameters_cc",
+    ":components_history_core_browser_features_cc",
+    ":components_history_clusters_core_features_cc",
+    ":components_history_clusters_core_on_device_clustering_features_cc",
     ":components_permissions_features_cc",
+    ":components_metrics_metrics_features_cc",
+    ":components_metrics_structured_structured_metrics_features_cc",
+    ":components_network_time_network_time_tracker_cc",
     ":components_offline_pages_core_offline_page_feature_cc",
     ":components_offline_pages_core_offline_page_feature_h",
+    ":components_omnibox_common_omnibox_features_cc",
+    ":components_optimization_guide_core_optimization_guide_features_cc",
+    ":components_page_image_service_features_cc",
+    ":components_password_manager_core_common_password_manager_features_cc",
+    ":components_privacy_sandbox_privacy_sandbox_features_cc",
+    ":components_safe_browsing_core_common_features_cc",
+    ":components_search_ntp_features_cc",
+    ":components_segmentation_platform_public_features_cc",
+    ":components_shared_highlighting_core_common_shared_highlighting_features_cc",
+    ":components_subresource_filter_core_common_common_features_cc",
     ":media_base_media_switches_cc",
     ":media_base_media_switches_h",
     ":net_base_features_cc",
     ":net_base_features_h",
     ":chrome_common_chrome_features_cc",
+    ":chrome_common_companion_visual_query_features_cc",
     ":chrome_browser_about_flags_cc",
     ":chrome_browser_flags_android_chrome_feature_list_cc",
     ":chrome_browser_flags_android_chrome_feature_list_h",
     ":chrome_browser_ui_ui_features_cc",
     ":chrome_browser_browser_features_cc",
+    ":chrome_browser_companion_core_features_cc",
     ":services_network_public_cpp_features_cc",
     ":services_network_public_cpp_features_h",
     ":third_party_blink_common_features_cc",
diff --git a/cromite_flags/android_webview/common/aw_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/android_webview/common/aw_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/android_webview/common/aw_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kWebViewEnumerateDevicesCache);
diff --git a/cromite_flags/android_webview/common/aw_features_cc/placeholder.txt b/cromite_flags/android_webview/common/aw_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/android_webview/common/aw_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/base/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/base/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/base/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kCollectAndroidFrameTimelineMetrics);
diff --git a/cromite_flags/base/features_cc/placeholder.txt b/cromite_flags/base/features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/base/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/chrome/browser/browser_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/chrome/browser/browser_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/chrome/browser/browser_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,4 @@
+SET_CROMITE_FEATURE_DISABLED(kDestroyProfileOnBrowserClose);
+SET_CROMITE_FEATURE_DISABLED(kFedCmWithoutThirdPartyCookies);
+SET_CROMITE_FEATURE_DISABLED(kOmniboxTriggerForNoStatePrefetch);
+// in Enable-Certificate-Transparency.inc: SET_CROMITE_FEATURE_ENABLED(kCertificateTransparencyAskBeforeEnabling);
diff --git a/cromite_flags/chrome/browser/companion/core/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/chrome/browser/companion/core/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/chrome/browser/companion/core/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,3 @@
+SET_CROMITE_FEATURE_DISABLED(kCompanionEnabledByObservingExpsNavigations);
+SET_CROMITE_FEATURE_DISABLED(kSidePanelCompanion);
+SET_CROMITE_FEATURE_DISABLED(kSidePanelCompanion2);
diff --git a/cromite_flags/chrome/browser/companion/core/features_cc/placeholder.txt b/cromite_flags/chrome/browser/companion/core/features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/chrome/browser/companion/core/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/chrome/browser/ui/ui_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/chrome/browser/ui/ui_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/chrome/browser/ui/ui_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+// in Disable-all-promo-dialogs.inc: SET_CROMITE_FEATURE_DISABLED(kChromeLabs);
diff --git a/cromite_flags/chrome/common/chrome_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/chrome/common/chrome_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/chrome/common/chrome_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,3 @@
+SET_CROMITE_FEATURE_DISABLED(kControlledFrame);
+SET_CROMITE_FEATURE_DISABLED(kKAnonymityService);
+SET_CROMITE_FEATURE_DISABLED(kKAnonymityServiceOHTTPRequests);
diff --git a/cromite_flags/chrome/common/companion/visual_query/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/chrome/common/companion/visual_query/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/chrome/common/companion/visual_query/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kVisualQuerySuggestions);
diff --git a/cromite_flags/chrome/common/companion/visual_query/features_cc/placeholder.txt b/cromite_flags/chrome/common/companion/visual_query/features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/chrome/common/companion/visual_query/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/aggregation_service/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/aggregation_service/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/aggregation_service/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kAggregationServiceMultipleCloudProviders);
diff --git a/cromite_flags/components/aggregation_service/features_cc/placeholder.txt b/cromite_flags/components/aggregation_service/features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/aggregation_service/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/autofill/core/common/autofill_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/autofill/core/common/autofill_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/autofill/core/common/autofill_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,2 @@
+SET_CROMITE_FEATURE_DISABLED(kAutofillServerCommunication);
+SET_CROMITE_FEATURE_ENABLED(kAutofillDisableShadowHeuristics);
diff --git a/cromite_flags/components/autofill/core/common/autofill_features_cc/placeholder.txt b/cromite_flags/components/autofill/core/common/autofill_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/autofill/core/common/autofill_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/autofill/core/common/autofill_payments_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/autofill/core/common/autofill_payments_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/autofill/core/common/autofill_payments_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kAutofillEnableRemadeDownstreamMetrics);
diff --git a/cromite_flags/components/autofill/core/common/autofill_payments_features_cc/placeholder.txt b/cromite_flags/components/autofill/core/common/autofill_payments_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/autofill/core/common/autofill_payments_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/commerce/core/commerce_feature_list_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/commerce/core/commerce_feature_list_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/commerce/core/commerce_feature_list_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,7 @@
+SET_CROMITE_FEATURE_DISABLED(kCommerceAllowOnDemandBookmarkUpdates);
+SET_CROMITE_FEATURE_DISABLED(kCommerceDeveloper);
+SET_CROMITE_FEATURE_DISABLED(kCommerceMerchantViewer);
+SET_CROMITE_FEATURE_DISABLED(kCommercePriceTracking);
+SET_CROMITE_FEATURE_DISABLED(kRetailCoupons);
+SET_CROMITE_FEATURE_DISABLED(kShoppingList);
+SET_CROMITE_FEATURE_DISABLED(kShoppingPDPMetrics);
diff --git a/cromite_flags/components/commerce/core/commerce_feature_list_cc/placeholder.txt b/cromite_flags/components/commerce/core/commerce_feature_list_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/commerce/core/commerce_feature_list_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/content_settings/core/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/content_settings/core/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/content_settings/core/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kUserBypassUI);
diff --git a/cromite_flags/components/heap_profiling/in_process/heap_profile_parameters_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/heap_profiling/in_process/heap_profile_parameters_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/heap_profiling/in_process/heap_profile_parameters_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kHeapProfilerReporting);
diff --git a/cromite_flags/components/heap_profiling/in_process/heap_profile_parameters_cc/placeholder.txt b/cromite_flags/components/heap_profiling/in_process/heap_profile_parameters_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/heap_profiling/in_process/heap_profile_parameters_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/history/core/browser/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/history/core/browser/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/history/core/browser/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kSyncSegmentsData);
diff --git a/cromite_flags/components/history/core/browser/features_cc/placeholder.txt b/cromite_flags/components/history/core/browser/features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/history/core/browser/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/history_clusters/core/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/history_clusters/core/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/history_clusters/core/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,11 @@
+SET_CROMITE_FEATURE_DISABLED(kHistoryClustersInternalsPage);
+SET_CROMITE_FEATURE_DISABLED(kHistoryClustersNavigationContextClustering);
+SET_CROMITE_FEATURE_DISABLED(kJourneys);
+SET_CROMITE_FEATURE_DISABLED(kJourneysNamedNewTabGroups);
+SET_CROMITE_FEATURE_DISABLED(kJourneysPersistCachesToPrefs);
+SET_CROMITE_FEATURE_DISABLED(kJourneysZeroStateFiltering);
+SET_CROMITE_FEATURE_DISABLED(kOmniboxAction);
+SET_CROMITE_FEATURE_DISABLED(kOmniboxHistoryClusterProvider);
+SET_CROMITE_FEATURE_DISABLED(kPersistContextAnnotationsInHistoryDb);
+SET_CROMITE_FEATURE_DISABLED(kPersistedClusters);
+SET_CROMITE_FEATURE_DISABLED(kSidePanelJourneys);
diff --git a/cromite_flags/components/history_clusters/core/features_cc/placeholder.txt b/cromite_flags/components/history_clusters/core/features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/history_clusters/core/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/history_clusters/core/on_device_clustering_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/history_clusters/core/on_device_clustering_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/history_clusters/core/on_device_clustering_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,2 @@
+SET_CROMITE_FEATURE_DISABLED(kOnDeviceClustering);
+SET_CROMITE_FEATURE_DISABLED(kOnDeviceClusteringKeywordFiltering);
diff --git a/cromite_flags/components/history_clusters/core/on_device_clustering_features_cc/placeholder.txt b/cromite_flags/components/history_clusters/core/on_device_clustering_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/history_clusters/core/on_device_clustering_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/metrics/metrics_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/metrics/metrics_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/metrics/metrics_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kStructuredMetrics);
diff --git a/cromite_flags/components/metrics/metrics_features_cc/placeholder.txt b/cromite_flags/components/metrics/metrics_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/metrics/metrics_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/metrics/structured/structured_metrics_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/metrics/structured/structured_metrics_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/metrics/structured/structured_metrics_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,3 @@
+SET_CROMITE_FEATURE_DISABLED(kEnabledStructuredMetricsService);
+SET_CROMITE_FEATURE_DISABLED(kNearbyShareMetrics);
+SET_CROMITE_FEATURE_DISABLED(kPhoneHubStructuredMetrics);
diff --git a/cromite_flags/components/metrics/structured/structured_metrics_features_cc/placeholder.txt b/cromite_flags/components/metrics/structured/structured_metrics_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/metrics/structured/structured_metrics_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/network_time/network_time_tracker_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/network_time/network_time_tracker_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/network_time/network_time_tracker_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kNetworkTimeServiceQuerying);
diff --git a/cromite_flags/components/network_time/network_time_tracker_cc/placeholder.txt b/cromite_flags/components/network_time/network_time_tracker_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/network_time/network_time_tracker_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/omnibox/common/omnibox_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/omnibox/common/omnibox_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/omnibox/common/omnibox_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,5 @@
+SET_CROMITE_FEATURE_DISABLED(kDocumentProviderNoSetting);
+SET_CROMITE_FEATURE_DISABLED(kDocumentProviderNoSyncRequirement);
+SET_CROMITE_FEATURE_DISABLED(kMlUrlScoring);
+SET_CROMITE_FEATURE_DISABLED(kRichAutocompletion);
+SET_CROMITE_FEATURE_DISABLED(kZeroSuggestPrefetching);
diff --git a/cromite_flags/components/omnibox/common/omnibox_features_cc/placeholder.txt b/cromite_flags/components/omnibox/common/omnibox_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/omnibox/common/omnibox_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/optimization_guide/core/optimization_guide_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/optimization_guide/core/optimization_guide_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/optimization_guide/core/optimization_guide_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kOptimizationGuidePersonalizedFetching);
diff --git a/cromite_flags/components/optimization_guide/core/optimization_guide_features_cc/placeholder.txt b/cromite_flags/components/optimization_guide/core/optimization_guide_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/optimization_guide/core/optimization_guide_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/page_image_service/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/page_image_service/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/page_image_service/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,2 @@
+SET_CROMITE_FEATURE_DISABLED(kImageService);
+SET_CROMITE_FEATURE_DISABLED(kImageServiceSuggestPoweredImages);
diff --git a/cromite_flags/components/page_image_service/features_cc/placeholder.txt b/cromite_flags/components/page_image_service/features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/page_image_service/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/password_manager/core/common/password_manager_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/password_manager/core/common/password_manager_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/password_manager/core/common/password_manager_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kUnifiedPasswordManagerLocalPasswordsMigrationWarning);
diff --git a/cromite_flags/components/password_manager/core/common/password_manager_features_cc/placeholder.txt b/cromite_flags/components/password_manager/core/common/password_manager_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/password_manager/core/common/password_manager_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/permissions/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/permissions/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/permissions/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kShowRelatedWebsiteSetsPermissionGrants);
diff --git a/cromite_flags/components/privacy_sandbox/privacy_sandbox_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/privacy_sandbox/privacy_sandbox_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/privacy_sandbox/privacy_sandbox_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,4 @@
+SET_CROMITE_FEATURE_DISABLED(kCookieSettingsUiAlignment);
+SET_CROMITE_FEATURE_DISABLED(kOverridePrivacySandboxSettingsLocalTesting);
+SET_CROMITE_FEATURE_DISABLED(kPrivacySandboxFirstPartySetsUI);
+SET_CROMITE_FEATURE_DISABLED(kPrivacySandboxProactiveTopicsBlocking);
diff --git a/cromite_flags/components/privacy_sandbox/privacy_sandbox_features_cc/placeholder.txt b/cromite_flags/components/privacy_sandbox/privacy_sandbox_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/privacy_sandbox/privacy_sandbox_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/safe_browsing/core/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/safe_browsing/core/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/safe_browsing/core/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,2 @@
+SET_CROMITE_FEATURE_DISABLED(kExtensionTelemetryTabsApiSignal);
+SET_CROMITE_FEATURE_DISABLED(kExtensionTelemetryDisableOffstoreExtensions);
diff --git a/cromite_flags/components/safe_browsing/core/common/features_cc/placeholder.txt b/cromite_flags/components/safe_browsing/core/common/features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/safe_browsing/core/common/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/search/ntp_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/search/ntp_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/search/ntp_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,6 @@
+SET_CROMITE_FEATURE_DISABLED(kCustomizeChromeSidePanelExtensionsCard);
+SET_CROMITE_FEATURE_DISABLED(kCustomizeChromeWallpaperSearch);
+SET_CROMITE_FEATURE_DISABLED(kNtpAlphaBackgroundCollections);
+SET_CROMITE_FEATURE_DISABLED(kNtpBackgroundImageErrorDetection);
+SET_CROMITE_FEATURE_DISABLED(kNtpChromeCartModule);
+SET_CROMITE_FEATURE_DISABLED(kNtpModulesMaxColumnCount);
diff --git a/cromite_flags/components/search/ntp_features_cc/placeholder.txt b/cromite_flags/components/search/ntp_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/search/ntp_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/segmentation_platform/public/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/segmentation_platform/public/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/segmentation_platform/public/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,3 @@
+SET_CROMITE_FEATURE_DISABLED(kSegmentationPlatformCollectTabRankData);
+SET_CROMITE_FEATURE_DISABLED(kSegmentationPlatformDeviceTier);
+SET_CROMITE_FEATURE_DISABLED(kSegmentationPlatformTimeDelaySampling);
diff --git a/cromite_flags/components/segmentation_platform/public/features_cc/placeholder.txt b/cromite_flags/components/segmentation_platform/public/features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/segmentation_platform/public/features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/shared_highlighting/core/common/shared_highlighting_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/shared_highlighting/core/common/shared_highlighting_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/shared_highlighting/core/common/shared_highlighting_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kSharedHighlightingManager);
diff --git a/cromite_flags/components/shared_highlighting/core/common/shared_highlighting_features_cc/placeholder.txt b/cromite_flags/components/shared_highlighting/core/common/shared_highlighting_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/shared_highlighting/core/common/shared_highlighting_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/components/subresource_filter/core/common/common_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/components/subresource_filter/core/common/common_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/components/subresource_filter/core/common/common_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kAdTagging);
diff --git a/cromite_flags/components/subresource_filter/core/common/common_features_cc/placeholder.txt b/cromite_flags/components/subresource_filter/core/common/common_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/components/subresource_filter/core/common/common_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/content/browser/preloading/prefetch/prefetch_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/content/browser/preloading/prefetch/prefetch_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/content/browser/preloading/prefetch/prefetch_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kPrefetchClientHints);
diff --git a/cromite_flags/content/browser/preloading/prefetch/prefetch_features_cc/placeholder.txt b/cromite_flags/content/browser/preloading/prefetch/prefetch_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/content/browser/preloading/prefetch/prefetch_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/cromite_flags/content/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/content/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/content/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,2 @@
+SET_CROMITE_FEATURE_DISABLED(kResourceTimingForCancelledNavigationInFrame);
+SET_CROMITE_FEATURE_DISABLED(kServiceWorkerAutoPreload);
diff --git a/cromite_flags/content/public/common/content_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/content/public/common/content_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/content/public/common/content_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,8 @@
+SET_CROMITE_FEATURE_DISABLED(kCookieDeprecationFacilitatedTesting);
+SET_CROMITE_FEATURE_DISABLED(kDigitalGoodsApi);
+SET_CROMITE_FEATURE_DISABLED(kDIPS);
+SET_CROMITE_FEATURE_DISABLED(kLegacyTechReportEnableCookieIssueReports);
+SET_CROMITE_FEATURE_DISABLED(kNotificationTriggers);
+SET_CROMITE_FEATURE_DISABLED(kWebIdentityDigitalCredentials);
+SET_CROMITE_FEATURE_DISABLED(kWebNfc);
+SET_CROMITE_FEATURE_DISABLED(kWebOTP);
diff --git a/cromite_flags/net/base/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/net/base/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/net/base/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,3 @@
+SET_CROMITE_FEATURE_DISABLED(kEnableWebTransportDraft07);
+SET_CROMITE_FEATURE_DISABLED(kTpcdMetadataGrants);
+SET_CROMITE_FEATURE_DISABLED(kWaitForFirstPartySetsInit);
diff --git a/cromite_flags/services/network/public/cpp/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/services/network/public/cpp/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/services/network/public/cpp/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,2 @@
+// in Disable-PrivateStateTokens-API.inc: SET_CROMITE_FEATURE_DISABLED(kFledgePst);
+// in Disable-PrivateStateTokens-API.inc: SET_CROMITE_FEATURE_DISABLED(kPrivateStateTokens);
diff --git a/cromite_flags/third_party/blink/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/third_party/blink/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/third_party/blink/common/features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1,18 @@
+SET_CROMITE_FEATURE_DISABLED(kAllowURNsInIframes);
+// in Keep-disabled-FetchLaterAPI.inc: SET_CROMITE_FEATURE_DISABLED(kAttributionReportingInBrowserMigration);
+SET_CROMITE_FEATURE_DISABLED(kBackgroundResourceFetch);
+SET_CROMITE_FEATURE_DISABLED(kBiddingAndScoringDebugReportingAPI);
+SET_CROMITE_FEATURE_DISABLED(kClientHintsFormFactors);
+SET_CROMITE_FEATURE_DISABLED(kFencedFramesM120FeaturesPart2);
+SET_CROMITE_FEATURE_DISABLED(kFledge);
+SET_CROMITE_FEATURE_DISABLED(kFledgeConsiderKAnonymity);
+SET_CROMITE_FEATURE_DISABLED(kFledgeEnforceKAnonymity);
+SET_CROMITE_FEATURE_DISABLED(kPrivateAggregationApiMultipleCloudProviders);
+SET_CROMITE_FEATURE_DISABLED(kSharedStorageAPI);
+SET_CROMITE_FEATURE_DISABLED(kSharedStorageAPIM118);
+SET_CROMITE_FEATURE_DISABLED(kSharedStorageAPIM124);
+SET_CROMITE_FEATURE_DISABLED(kSharedStorageSelectURLLimit);
+SET_CROMITE_FEATURE_DISABLED(kSpeculationRulesPrefetchFuture);
+SET_CROMITE_FEATURE_ENABLED(kMixedContentAutoupgrade);
+SET_CROMITE_FEATURE_ENABLED(kReducedReferrerGranularity);
+SET_CROMITE_FEATURE_ENABLED(kUACHOverrideBlank);
diff --git a/cromite_flags/ui/base/ui_base_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc b/cromite_flags/ui/base/ui_base_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/ui/base/ui_base_features_cc/Bring-in-Brave-feature-states-for-privacy-security.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kChromeRefresh2023);
diff --git a/cromite_flags/ui/base/ui_base_features_cc/placeholder.txt b/cromite_flags/ui/base/ui_base_features_cc/placeholder.txt
new file mode 100755
--- /dev/null
+++ b/cromite_flags/ui/base/ui_base_features_cc/placeholder.txt
@@ -0,0 +1 @@
+this file is intentionally empty
diff --git a/third_party/blink/renderer/core/html/html_script_element.cc b/third_party/blink/renderer/core/html/html_script_element.cc
--- a/third_party/blink/renderer/core/html/html_script_element.cc
+++ b/third_party/blink/renderer/core/html/html_script_element.cc
@@ -375,8 +375,8 @@ bool HTMLScriptElement::supports(const AtomicString& type) {
   if (type == script_type_names::kImportmap)
     return true;
 
-  if (type == script_type_names::kSpeculationrules) {
-    return true;
+  if (type == script_type_names::kSpeculationrules) {  // disabled
+    return false;                                      // in Brave
   }
   if (type == script_type_names::kWebbundle)
     return true;
diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -263,15 +263,15 @@
     },
     {
       // Interest Group JS API/runtimeflag.
-      name: "AdInterestGroupAPI",
+      name: "AdInterestGroupAPI", // disabled
+      status: {},                 // in Brave
       origin_trial_feature_name: "AdInterestGroupAPI",
-      implied_by: ["Fledge", "Parakeet"],
       public: true,
     },
     {
-      name: "AdTagging",
+      name: "AdTagging", // disabled
+      status: {},        // in Brave
       public: true,
-      status: "test",
       base_feature: "none",
     },
     {
@@ -309,8 +309,10 @@
       name: "AnimationWorklet",
     },
     {
-      name: "AnonymousIframe",
-      status: "stable",
+      name: "AnonymousIframe", // disabled
+      status: {},              // in Brave
+      public: true,
+      base_feature: "none",
     },
     {
       name: "AOMAriaRelationshipProperties",
@@ -686,12 +688,9 @@
     {
       // blink::features::kComputePressure is a kill switch for the API. If the
       // feature is disabled, origin trial tokens are ignored.
-      name: "ComputePressure",
+      name: "ComputePressure", // disabled
+      status: {},              // in Brave
       origin_trial_feature_name: "ComputePressure_v2",
-      origin_trial_allows_third_party: true,
-      status: "experimental",
-      base_feature_status: "enabled",
-      copied_from_base_feature_if: "overridden",
     },
     {
       name: "ConsolidatedMovementXY",
@@ -1069,8 +1068,8 @@
       implied_by: ["CSSStickyContainerQueries", "CSSStickyContainerQueries"],
     },
     {
-      name: "CSSSelectorFragmentAnchor",
-      status: "experimental",
+      name: "CSSSelectorFragmentAnchor", // disabled
+      status: {},                        // in Brave
       base_feature: "CssSelectorFragmentAnchor",
     },
     {
@@ -1791,7 +1790,8 @@
       status: "experimental",
     },
     {
-      name: "Fledge",
+      name: "Fledge", // disabled
+      status: {},     // in Brave
       base_feature: "none",
       origin_trial_feature_name: "PrivacySandboxAdsAPIs",
       origin_trial_allows_third_party: true,
@@ -1833,7 +1833,8 @@
       base_feature: "none",
     },
     {
-      name: "FledgeMultiBid",
+      name: "FledgeMultiBid", // disabled
+      status: {},             // in Brave
       public: true,
     },
     {
@@ -2869,7 +2870,8 @@
     },
     {
       // PARAKEET ad serving runtime flag/JS API.
-      name: "Parakeet",
+      name: "Parakeet", // disabled
+      status: {},       // in Brave
       origin_trial_feature_name: "Parakeet",
     },
     // This is to add an option to enable the Reveal button on password inputs while waiting ::reveal gets standardized.
@@ -3040,8 +3042,8 @@
       // preloading.) See https://crbug.com/1494471.
       //
       // It also has some feature params defined throughout the codebase.
-      name: "Prerender2",
-      status: "test",
+      name: "Prerender2", // disabled
+      status: {},         // in Brave
     },
     {
       name: "Presentation",
@@ -4312,8 +4314,8 @@
     {
       // WebGPU experimental features are meant for features that would
       // eventually land in the WebGPU spec.
-      name: "WebGPUExperimentalFeatures",
-      status: "experimental",
+      name: "WebGPUExperimentalFeatures", // disabled
+      status: {},                         // in Brave
       public: true,
     },
     {
-- 

